### PR TITLE
wishlist

### PR DESCRIPTION
**Pull request by arijit1111**
add to wishlist, fetch wishlist

### CODE REVIEW

**Comment by properweb** [comment]
Task linked: [CU-2yehr3e Customer Wishlist](https://app.clickup.com/t/2yehr3e)

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+                    $tprice = 0;
+                    foreach ($qtyArr as $qk => $qv) {
+                        $tprice += $priceArr[$qk] * $qv;
+                    }
+                    $alreadyWished->price = $tprice;
+                    $alreadyWished->style_name = rtrim(implode(',', $styleArr), ',');
+                    $alreadyWished->style_group_name = rtrim(implode(',', $styleGrpArr), ',');
+                    $alreadyWished->reference = serialize($optionsArr);
+                }
+                if ($productType == 'PREPACK') {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php]: Replace == with ===

see https://www.php.net/manual/en/language.operators.comparison.php

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+                    $prepackVariant = ProductPrepack::where('id', $request->prepack_id)->first();
+                    $alreadyWished->price = $prepackVariant->packs_price;
+                    $alreadyWished->style_name = $prepackVariant->style;
+                    $alreadyWished->style_group_name = $prepackVariant->size_ratio . ';' . $prepackVariant->size_range;
+                }
+                if ($productType == 'SINGLE_PRODUCT') {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php]: Replace == with ===

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+            } else {
+                if ($variant->stock < $alreadyWished->quantity || $variant->stock <= 0) {
+                    $response = ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+                    return response()->json($response);
+                }
+            }
+            $alreadyWished->save();
+        } else {
+            $boardCount = Board::where('user_id', $request->user_id)->get()->count();
+            if ($boardCount == 0) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php]: Replace == with ===

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 4]
```diff
@@ -257,16 +266,19 @@ public function updateShop(Request $request)
...
+    public function goLive(Request $request) {
 
         $brand = Brand::where('user_id', $request->user_id)->first();
         if ($brand) {
             $user = User::where('id', $request->user_id)->first();
             if ($user && $user->verified == '1') {
                 $token = Str::random(64);
                 $brandName = $brand->brand_name;
-
+//                Mail::send('email.goLive', ['site_url' => 'https://demoupdates.com/updates/new-bazar/dev/', 'site_name' => 'BAZAR', 'name' => $brand_name], function($message) use($request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove the commented code if not needed

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 5]
```diff
@@ -439,12 +448,15 @@ public function order(Request $request, $order_number)
             );
         }
 
+//        echo '<pre>';
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove the commented code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 6]
```diff
@@ -496,23 +508,26 @@ public function ordersPackingSlip(Request $request)
             }
         }
 
+//        echo '<pre>';
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: remove commented code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 7]
```diff
@@ -535,17 +550,20 @@ public function acceptOrder(Request $request)
             $brand = Brand::where('user_id', $order->brand_id)->first();
             $msg = "Your orders with " . $brand->brand_name . " having order number <strong>#" . $order->order_number . "</strong> has been processing.";
             $data = array('email' => $retailer_user->email, 'order_number' => $order->order_number);
-
+//            Mail::send('email.orderStatus', ['msg' => $msg, 'site_url' => 'https://demoupdates.com/updates/new-bazar/dev/', 'site_name' => 'BAZAR', 'name' => $retailer_user->first_name . ' ' . $retailer_user->last_name], function($message) use($data) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: remove commented code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 8]
```diff
@@ -561,25 +579,31 @@ public function changeDateOrder(Request $request)
                     $retailer_user = User::find($retailerId);
                     $msg = "Your order's ship date with " . $brand->brand_name . " having order number <strong>#" . $order->order_number . "</strong> has been changed to " . $request->ship_date;
                     $data = array('email' => $retailer_user->email, 'order_number' => $order->order_number);
-
-
+//                    Mail::send('email.orderStatus', ['msg' => $msg, 'site_url' => 'https://demoupdates.com/updates/new-bazar/dev/', 'site_name' => 'BAZAR', 'name' => $retailer_user->first_name . ' ' . $retailer_user->last_name], function($message) use($data) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: remove commented code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 9]
```diff
@@ -19,35 +20,31 @@
...
-class BrandController extends Controller
-{
+class BrandController extends Controller {
 
     private $brandAbsPath = "";
     private $brandRelPath = "";
 
-    public function __construct()
-    {
+    public function __construct() {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 10]
```diff
@@ -19,35 +20,31 @@
...
         $this->brandAbsPath = public_path('uploads/brands');
         $this->brandRelPath = 'uploads/brands/';
     }
 
-    public function index()
-    {
+    public function index() {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 11]
```diff
@@ -19,35 +20,31 @@
...
         return view('brand::index');
     }
 
-    public function register(Request $request)
-    {
+    public function register(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 12]
```diff
@@ -68,9 +65,8 @@ public function register(Request $request)
         return response()->json($response);
     }
 
-    public function create(Request $request)
-    {
-        $data = (array)$request->all();
+    public function create(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 13]
```diff
@@ -257,16 +266,19 @@ public function updateShop(Request $request)
         return response()->json($response);
     }
 
-    public function goLive(Request $request)
-    {
+    public function goLive(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 14]
```diff
@@ -280,8 +292,7 @@ public function goLive(Request $request)
         return response()->json($response);
     }
 
-    public function all($id)
-    {
+    public function all($id) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 15]
```diff
@@ -359,8 +369,7 @@ public function orders(Request $request)
         }
     }
 
-    public function order(Request $request, $order_number)
-    {
+    public function order(Request $request, $order_number) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: What's the purpose of the order function override?

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 16]
```diff
@@ -301,8 +312,7 @@ public function all($id)
         return response()->json($response);
     }
 
-    public function orders(Request $request)
-    {
+    public function orders(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing, and what does this function do? if it's returning customer orders then this function should be in the orders module, not in the brands section

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 17]
```diff
@@ -439,12 +448,15 @@ public function order(Request $request, $order_number)
...
+//        print_r($cart_arr);
+//        exit;
+        //dd($user);
         $response = ['res' => true, 'msg' => "", 'data' => $data];
         return response()->json($response);
     }
 
-    public function ordersPackingSlip(Request $request)
-    {
+    public function ordersPackingSlip(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: All order related functions should be moved to the order/sales module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 18]
```diff
@@ -496,23 +508,26 @@ public function ordersPackingSlip(Request $request)
...
+//        print_r($cart_arr);
+//        exit;
+        //dd($user);
         $response = ['res' => true, 'msg' => "", 'data' => $data];
         return response()->json($response);
     }
 
-    public function acceptOrder(Request $request)
-    {
+    public function acceptOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: All order related functions should be moved to the order/sales module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 19]
```diff
@@ -535,17 +550,20 @@ public function acceptOrder(Request $request)
...
-
+            // dd($users);        
             $response = ['res' => true, 'msg' => '', 'data' => $data];
         }
         return response()->json($response);
     }
 
-    public function changeDateOrder(Request $request)
-    {
+    public function changeDateOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: All order related functions should be moved to the order/sales module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 20]
```diff
@@ -561,25 +579,31 @@ public function changeDateOrder(Request $request)
...
+//        print_r($cart_arr);
+//        exit;
+        //dd($user);
         $response = ['res' => true, 'msg' => "", 'data' => $data];
         return response()->json($response);
     }
 
-    public function changeAddressOrder(Request $request)
-    {
+    public function changeAddressOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: All order related functions should be moved to the order/sales module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 21]
```diff
@@ -600,8 +624,7 @@ public function changeAddressOrder(Request $request)
         return response()->json($response);
     }
 
-    public function cancelOrder(Request $request)
-    {
+    public function cancelOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: All order related functions should be moved to the order/sales module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 22]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
             $order->save();
             $brand = Brand::where('user_id', $order->brand_id)->first();
             $prdct_arr = Cart::where('order_id', $order->id)->get();
-
+            //sync stock to external sites
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove commented out code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 23]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
 
             $retailerId = $order->user_id;
             $retailer_user = User::find($retailerId);
             $msg = "Your order with " . $brand->brand_name . " having order number <strong>#" . $order->order_number . "</strong> has been cancelled.<br>";
-            $msg .= "<strong>Reason for cancelling</strong><br>";
+            $msg .="<strong>Reason for cancelling</strong><br>";
             $msg .= $request->cancel_reason_title . "<br>";
             $msg .= $request->cancel_reason_desc . "<br>";
             $data = array('email' => $retailer_user->email, 'order_number' => $order->order_number);
+//            Mail::send('email.orderStatus', ['msg' => $msg, 'site_url' => 'https://demoupdates.com/updates/new-bazar/dev/', 'site_name' => 'BAZAR', 'name' => $retailer_user->first_name . ' ' . $retailer_user->last_name], function($message) use($data) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove commented out code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 24]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+                $product = Products::where('id', $prdct->product_id)->first();
+
+                $syncs = DB::table('brand_store_import_tbl')
+                                ->where('brand_id', $brand_id)
+                                ->where('website', $product->website)
+                                ->get()->first();
+
+                $types = $syncs->types;
+                if ($types == 'wordpress') {
+                    // include(app_path() . '/Classes/class-sw-api-client.php');
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove commented out code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 25]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+
+
+                    $consumer_key = $syncs->api_key;
+
+                    $website = 'https://' . $syncs->website;
+                    $consumer_secret = $syncs->api_password;
+
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: remove blank lines

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 26]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+                    $url = "" . $website . "/wp-json/wc/v3/products/" . $prdct_res[0]->product_id . "";
+
+                    $headers = array(
+                        'Authorization' => 'Basic ' . base64_encode($consumer_key . ':' . $consumer_secret)
+                    );
+                    $data = array(
+                        'stock_quantity' => $prdct_res[0]->stock,
+                    );
+
+                    $curl = curl_init();
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: move curl commands into a separate function, or even better if you create a base module in which you can list all common helper functions that can be used across the site.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 27]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+//for debug only!
+                    curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
+                    curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
+                    curl_setopt($curl, CURLOPT_USERPWD, "$consumer_key:$consumer_secret");
+                    $resp = curl_exec($curl);
+                    $status_code = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+                    curl_close($curl);
+
+
+                    $qry = "SELECT * FROM product_variations WHERE website='" . $syncs->website . "' AND product_id='" . $product->product_id . "'";
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove raw SQL and replace it with proper format

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 28]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+                            $url = "" . $website . "/wp-json/wc/v3/products/" . $prdct_res[0]->product_id . "/variations/" . $var->variation_id . "";
+
+                            $headers = array(
+                                'Authorization' => 'Basic ' . base64_encode($consumer_key . ':' . $consumer_secret)
+                            );
+                            $data = array(
+                                'stock_quantity' => $var->stock,
+                            );
+
+                            $curl = curl_init();
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Duplicate code. As mentioned, move curl into a separate function and create a base class that can be used across the site.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 29]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+//                $message->to($data['email']);
+//                $message->from("sender@demoupdates.com");
+//                $message->subject('Bazar:' . $data["order_number"] . ' Status');
+//            });
         }
         $response = ['res' => true, 'msg' => "", 'data' => ""];
         return response()->json($response);
     }
 
+    public function syncExternal($prdct_arr, $brand_id) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: What is this function supposed to do? this is clearly not the way we discussed things.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 30]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+
+                if ($types == 'shopify') {
+
+                    $API_KEY = $syncs->api_key;
+                    $STORE_URL = $syncs->website;
+                    $PASSWORD = $syncs->api_password;
+
+                    $putUrl = 'https://' . $API_KEY . ':' . $PASSWORD . '@' . $STORE_URL . '/admin/api/2022-07/inventory_levels/set.json';
+
+                    $qry = "SELECT * FROM product_variations WHERE website='" . $syncs->website . "' AND product_id='" . $product->id . "'";
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove raw SQL and replace with the proper format

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 31]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+                        foreach ($res as $var) {
+
+                            $payload = array(
+                                "location_id" => 36132814934,
+                                "inventory_item_id" => $var->inventory_item_id,
+                                "available" => $var->stock
+                            );
+                            $payload = json_encode($payload, JSON_NUMERIC_CHECK);
+
+                            $session = curl_init();
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Duplicate curl code.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 32]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+//curl_setopt($session, CURLOPT_PUT, 1); 
+                            curl_setopt($session, CURLOPT_POSTFIELDS, $payload);
+                            curl_setopt($session, CURLOPT_CUSTOMREQUEST, 'POST');
+                            curl_setopt($session, CURLOPT_HTTPHEADER, array('Accept: application/json', 'Content-Type: application/json'));
+                            $data = curl_exec($session);
+                            curl_close($session);
+                        }
+                    }
+
+                    $qry = "SELECT * FROM product_variations WHERE website='" . $syncs->website . "' AND product_id='" . $product->id . "'";
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Again, raw SQL found.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 33]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+
+
+                            $payload = array(
+                                "location_id" => 36132814934,
+                                "inventory_item_id" => $var->inventory_item_id,
+                                "available" => $prdct_res[0]->stock
+                            );
+                            $payload = json_encode($payload, JSON_NUMERIC_CHECK);
+
+                            $session = curl_init();
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: duplicate curl code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 34]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
+            }
+        }
+        return true;
+    }
+
+    public function splitOrder(Request $request) {
         $order = Order::find($request->order_id);
         $new_cart = $request->items;
-
+//        print_r($new_cart);
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: remove commented out code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 35]
```diff
@@ -611,26 +634,226 @@ public function cancelOrder(Request $request)
...
-    public function splitOrder(Request $request)
-    {
+                    //$response = ['res' => true, 'msg' => "Sync Successfully", 'data' => ""];
+                }
+            }
+        }
+        return true;
+    }
+
+    public function splitOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Move order related functions to order/sales module

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 36]
```diff
@@ -655,6 +878,8 @@ public function splitOrder(Request $request)
             $order->total_amount = $order->sub_total;
             $order->save();
 
+            //$cart = Cart::where('user_id', $order->user_id)->where('user_id', $order->brand_id)->where('order_id', null);
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: Remove commented out code

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 37]
```diff
@@ -668,28 +893,28 @@ public function splitOrder(Request $request)
         return response()->json($response);
     }
 
-    public function updateOrder(Request $request)
-    {
+    public function updateOrder(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing, and potentially a duplicate function.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 38]
```diff
@@ -700,8 +925,7 @@ public function updateOrder(Request $request)
         return response()->json($response);
     }
 
-    public function ordersCSV(Request $request)
-    {
+    public function ordersCSV(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing. Move all order related functions into sales/order module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 39]
```diff
@@ -722,9 +946,80 @@ public function ordersCSV(Request $request)
         exit();
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: why exit()? what's the purpose of this function?

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 40]
```diff
@@ -722,9 +946,80 @@ public function ordersCSV(Request $request)
...
     }
 
+    public function customers(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing and move all customer related functions into the customer module.

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 41]
```diff
@@ -722,9 +946,80 @@ public function ordersCSV(Request $request)
...
+            $data = $customers;
+            $response = ['res' => true, 'msg' => "", 'data' => $data];
+            return response()->json($response);
+        } else {
+            $response = ['res' => false, 'msg' => "No record found", 'data' => ""];
+            return response()->json($response);
+        }
+    }
+
+    public function addCustomer(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs missing and move all customer related functions into the customer module.

File: Modules/Wishlist/Entities/Board.php [hunk 42]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Modules\Wishlist\Entities;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+
+class Board extends Model {
+
+    protected $fillable = ['board_key','name', 'visibility'];
+
```
Comment (shoaibrehman) [inline Modules/Wishlist/Entities/Board.php:11]: Remove blank lines

File: Modules/Wishlist/Entities/Wishlist.php [hunk 43]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Modules\Wishlist\Entities;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+
+class Wishlist extends Model {
+
+    protected $fillable = ['user_id','brand_id', 'product_id', 'order_id', 'quantity', 'amount', 'price', 'status'];
+
```
Comment (shoaibrehman) [inline Modules/Wishlist/Entities/Wishlist.php:11]: Remove blank lines

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 44]
```diff
@@ -0,0 +1,502 @@
...
+use Modules\User\Entities\User;
+use Modules\Brand\Entities\Brand;
+use Modules\Product\Entities\Products;
+use Modules\Product\Entities\ProductVariation;
+use Modules\Product\Entities\ProductPrepack;
+use Modules\Country\Entities\Country;
+
+class WishlistController extends Controller {
+
+    public function add(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:25]: PHPDocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 45]
```diff
@@ -0,0 +1,502 @@
...
+                    $optionsArr = [];
+                    $styleArr = [];
+                    $styleGrpArr = [];
+                    $sizeArr = [];
+                    $priceArr = [];
+                    $qtyArr = [];
+
+                    if (!empty($request->openSizingArray)) {
+                        foreach ($request->openSizingArray as $size) {
+                            $sizeVariant = ProductVariation::where('product_id', $product->id)->where('options1', "Size")->where('value1', $size['value'])->where('value2', $variant->value2)->where('value3', $variant->value3)->first();
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:167]: Replace these Ifs with a switch statement as that's a better option performance wise. See https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#:~:text=Generally%20speaking%2C%20switch%20is%20always,using%20if%2Delse%20or%20switch%20.

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 46]
```diff
@@ -0,0 +1,502 @@
...
+            }
+            $wishlist->amount = $wishlist->price * $wishlist->quantity;
+            $wishlist->type = $productType;
+            $wishlist->save();
+        }
+        $response = ['res' => true, 'msg' => 'Product successfully added to cart', 'data' => ""];
+        return response()->json($response);
+    }
+
+    public function delete(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:252]: phpdocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 47]
```diff
@@ -0,0 +1,502 @@
...
+        if ($wishlist) {
+            $wishlist->delete();
+            $response = ['res' => true, 'msg' => 'Cart successfully removed', 'data' => ""];
+            return response()->json($response);
+        }
+        $response = ['res' => false, 'msg' => 'Error please try again', 'data' => ""];
+        return response()->json($response);
+    }
+
+    public function fetch(Request $request, $id) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:268]: phpdocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 48]
```diff
@@ -0,0 +1,502 @@
...
+            'board_arr' => $boardArr,
+            'brand_arr' => $brandArr,
+            'product_arr' => $productArr,
+        );
+
+        $response = ['res' => true, 'msg' => "", 'data' => $data];
+        return response()->json($response);
+    }
+
+    public function fetchBoards(Request $request, $id) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:367]: phpdocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 49]
```diff
@@ -0,0 +1,502 @@
...
+            }
+        }
+
+        $data = $boardArr;
+
+        $response = ['res' => true, 'msg' => "", 'data' => $data];
+        return response()->json($response);
+    }
+
+    public function fetchBoard(Request $request, $key) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:405]: phpdocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 50]
```diff
@@ -0,0 +1,502 @@
...
+        }
+        $data = array(
+            'board_arr' => $boardArr,
+            'product_arr' => $productArr,
+        );
+        $response = ['res' => true, 'msg' => "", 'data' => $data];
+        return response()->json($response);
+    }
+
+    public function addBoard(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:450]: phpdocs missing, and what is a board? Is it wishlish specific? Since you're add, updating and deleting boards so a better approach would be to create it as a separate module to make code more manageable.

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 51]
```diff
@@ -0,0 +1,502 @@
...
+                $response = ['res' => true, 'msg' => "Successfully added your board", 'data' => ''];
+            } else {
+                $response = ['res' => false, 'msg' => "Please try again!", 'data' => ''];
+            }
+        }
+
+        return response()->json($response);
+    }
+
+    public function updateBoard(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:477]: phpdocs missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 52]
```diff
@@ -0,0 +1,502 @@
...
+                $response = ['res' => true, 'msg' => "Successfully updated your board", 'data' => ''];
+            } else {
+                $response = ['res' => false, 'msg' => "Please try again!", 'data' => ''];
+            }
+        }
+
+        return response()->json($response);
+    }
+
+    public function deleteBoard(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:502]: phpdocs missing.

File: Modules/Wishlist/Http/Controllers/WishlistController.php [hunk 53]
```diff
@@ -0,0 +1,502 @@
...
+            }
+            Board::where('id', $board->id)->delete();
+            $response = ['res' => true, 'msg' => 'Board successfully removed', 'data' => ""];
+            return response()->json($response);
+        }
+        $response = ['res' => false, 'msg' => 'Error please try again', 'data' => ""];
+        return response()->json($response);
+    }
+
+    public function changeBoard(Request $request) {
```
Comment (shoaibrehman) [inline Modules/Wishlist/Http/Controllers/WishlistController.php:523]: phpdocs missing

File: Modules/Wishlist/Resources/views/index.blade.php [hunk 54]
```diff
@@ -0,0 +1,9 @@
+@extends('wishlist::layouts.master')
+
+@section('content')
+    <h1>Hello World</h1>
```
Comment (shoaibrehman) [inline Modules/Wishlist/Resources/views/index.blade.php:4]: Is this file needed? Remove if not.

**Review by shoaibrehman — CHANGES_REQUESTED** [review]
Please review the feedback and let's have a call to discuss it once you have reviewed it.

File: Modules/Wishlist/Http/Controllers/WishlistController.php (hunk 50 above)
Comment (arijit1111) [reply Modules/Wishlist/Http/Controllers/WishlistController.php:450]: wishlish specific

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 39 above)
Comment (arijit1111) [reply Modules/Brand/Http/Controllers/BrandController.php]: download csv file

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 10 above)
Comment (shoaibrehman) [reply Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs is still missing

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 11 above)
Comment (shoaibrehman) [reply Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs is still missing

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 12 above)
Comment (shoaibrehman) [reply Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs is still missing

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 13 above)
Comment (shoaibrehman) [reply Modules/Brand/Http/Controllers/BrandController.php]: PHPDocs is still missing

File: Modules/Retailer/Http/Controllers/RetailerController.php [hunk 55]
```diff
@@ -2,23 +2,20 @@
...
 use Illuminate\Support\Facades\Validator;
-use Modules\User\Entities\User;
 use Modules\Retailer\Entities\Retailer;
+use Modules\User\Entities\User;
 
 
 class RetailerController extends Controller
 {
 
     public function register(Request $request)
```
Comment (shoaibrehman) [inline Modules/Retailer/Http/Controllers/RetailerController.php:17]: PHPDocs is still missing

File: Modules/Retailer/Http/Controllers/RetailerController.php [hunk 56]
```diff
@@ -63,5 +58,70 @@ public function register(Request $request)
         return response()->json($response);
     }
 
+    public function edit($id)
```
Comment (shoaibrehman) [inline Modules/Retailer/Http/Controllers/RetailerController.php:61]: PHPDocs is still missing

File: Modules/Retailer/Http/Controllers/RetailerController.php [hunk 57]
```diff
@@ -63,5 +58,70 @@ public function register(Request $request)
...
+        $retailer->first_name = $user->first_name;
+        $retailer->last_name = $user->last_name;
+        $retailer->email = $user->email;
+        $retailer->verified = $user->verified;
+
+        $response = ['res' => true, 'msg' => "", 'data' => $retailer];
+        return response()->json($response);
+    }
+
+    public function update(Request $request)
```
Comment (shoaibrehman) [inline Modules/Retailer/Http/Controllers/RetailerController.php:75]: PHPDocs is still missing

File: Modules/Wishlist/Http/Controllers/WishlistController.php (hunk 45 above)
Comment (shoaibrehman) [reply Modules/Wishlist/Http/Controllers/WishlistController.php:167]: This is not yet complete?

**Review by shoaibrehman — CHANGES_REQUESTED** [review]
There are still issues pending. Please resolve and resubmit.

File: Modules/Wishlist/Http/Controllers/WishlistController.php (hunk 45 above)
Comment (arijit1111) [reply Modules/Wishlist/Http/Controllers/WishlistController.php:167]: Here switch statement is not possible. In this portion IF can not be replaced by switch

**Comment by SonaKhachatryan1995** [comment]
not actual anymore

**Comment by SonaKhachatryan1995** [comment]
not actual anymore
